homeassistant/components/tplink_omada/config_flow.py

```diff
@@ -0,0 +1,199 @@
+
+    client = await create_omada_client(hass, MappingProxyType(data))
+    controller_id = await client.login()
+    name = await client.get_controller_name()
+    sites = await client.get_sites()
+
+    return HubInfo(controller_id, name, sites)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for TP-Link Omada."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Create the config flow for a new integration."""
+        self._omada_opts: dict[str, Any] = {}
+        self._sites: list[OmadaSite] = []
+        self._controller_name = ""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+
+        errors: dict[str, str] = {}
+        info = None
+        if user_input is not None:
+            info = await self._test_login(user_input, errors)
+
+        if info is None or user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+            )
+
+        await self.async_set_unique_id(info.controller_id)
+        self._abort_if_unique_id_configured()
+
+        self._omada_opts.update(user_input)
+        self._sites = info.sites
+        self._controller_name = info.name
+        if len(self._sites) > 1:
+            return await self.async_step_site()
+        return await self.async_step_site({CONF_SITE: self._sites[0].id})
+
+    async def async_step_site(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle step to select site to manage."""
+
+        if user_input is None:
+            schema = vol.Schema(
+                {
+                    vol.Required(CONF_SITE, "site"): selector.SelectSelector(
+                        selector.SelectSelectorConfig(
+                            options=[
+                                selector.SelectOptionDict(value=s.id, label=s.name)
+                                for s in self._sites
+                            ],
+                            multiple=False,
+                            mode=selector.SelectSelectorMode.DROPDOWN,
+                        )
+                    )
+                }
+            )
+
+            return self.async_show_form(step_id="site", data_schema=schema)
+
+        self._omada_opts.update(user_input)
+        site_name = next(
+            site for site in self._sites if site.id == user_input["site"]
+        ).name
+        display_name = f"{self._controller_name} ({site_name})"
+
+        return self.async_create_entry(title=display_name, data=self._omada_opts)
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self._omada_opts = dict(entry_data)
+        return await self.async_step_reauth_confirm(dict(entry_data))
```

Comment:
Why pass the credentials as input to `reauth_confirm`? They are no longer valid, or else we wouldn't have gotten into `reauth` to begin with, so trying them again here will trigger an error again (and maybe count as a failed login attempt in case Omada keeps track).

Reply:
Good point. Not sure why I left that in there.